### ISSUE TITLE
Bug: Application crashing while checking `state.hasTag()` of the actor machines in Samsung 

### ISSUE THREAD

**Issue by jsgiant** [statelyai/xstate#3156]
### Description

We have a use case where we need to maintain child actors in a  parent machine and `hasTag` method is working fine for the parent machine state but the application is crashing when we try to use the same with the actor's state. The issue is getting only on specific mobile devices.

- The things are working as expected when we submit the form with keyboard events, but the issues are popping out when we try to submit the form on clicking the submit button that we kept in the form. 

### Expected result

Should work properly without any crashes

### Actual result

Application is crashing

### Reproduction

https://bit.ly/3JiqGqi

### Device Config

- Model: Samsung Galaxy M31
- OS Type: Android
- Browsers: Chrome, Samsung Internet
- Android version: 11

### Versions:
-  "xstate": "4.30.6",
-  "@xstate/react": "2.0.1",

#### NOTE:
- We have upgraded to the latest version from `1.19.1` but the issue is still persisting

**Comment by Andarist** [comment]
I didnt fully confirm this here but it looks like what i call a “deferred children problem”. Im already preparing a fix for this

**Comment by jsgiant** [comment]
Hi @Andarist, Thanks for the quick response. When can we expect the fix? We're using the XState with the same structure in multiple projects on production. Any quick workaround for this will be helpful for us.

Thanks.

**Comment by Andarist** [comment]
I thank that we should be able to release a fix next week, depending on your use case I might post a workaround for this a little bit later today (I'm on mobile right now and don't have access to that workaround now)

**Comment by davidkpiano** [comment]
I refactored this and was able to get it working. You can use _lazy context_ to achieve the same result:

```ts
const machine = createMachine({
  // ...
  context: () => ({
    actorRef: spawn(actorMachine)
  }),
  initial: "init",
  id: "(machine)",
  states: {/* ... */}
});
```

See here: https://codesandbox.io/s/boring-hermann-22ke90?file=/src/index.tsx

**Comment by jsgiant** [comment]
Hello @davidkpiano, We have a different use case here to spawn the actors based on the service response invoked in the flow. We can't spawn the actors using the lazy context as the actor that we need to spawn will be changed. 

See our use case here: https://codesandbox.io/s/dreamy-kirch-7vjxcr?file=/src/index.tsx

Thanks.

**Comment by Andarist** [comment]
Could you list your repro steps leading to a crash? I've checked the latest shared codesandbox and it looks OK at the runtime level:
<img width="603" alt="Screenshot 2022-03-17 at 12 35 09" src="https://user-images.githubusercontent.com/9800850/158800469-8db82ca2-77f8-403c-8269-956fac341dc0.png">

As to the code that has been initially shared here. It seems that, at least for this problem, the answer is to split components a little bit. You can't read the state of an actor that doesn't exist yet - so if it's initialized lazily, in response to an HTTP request or anything else, then you need to account for that and push your `useActor(lazyActorRef)` to some inner component that would be rendered conditionally. 

In theory, we could support `null/undefined` as an argument to `useActor`. However, that IMHO is not the way to go cause we need to ask ourselves a question - what you would get back as "state" of such an "actor"? And how to handle this at the type level? 

We can consider a simplified declaration for the `useActor` to be roughly equivalent to:
```ts
declare function useActor<T, TEvent>(
  actorRef: ActorRef<T, TEvent>
): [state: T, send: (ev: TEvent) => void];

```

With such a declaration the caller of `useActor` expects to get back `T` in return. But how can we possibly give back to the caller if `actorRef` is `null`/`undefined`? We just can't do that and I don't expect everyone to use defensive programming on the returned `state`, such as `state?.someValue`. That IMHO would make reasoning about the code harder.

We could consider changing the signature to be roughly equivalent to:
```ts
declare function useActor<T, TEvent>(actorRef: ActorRef<T, TEvent>, defaultValue: T): [state: T, send: (ev: TEvent) => void]
```
But then you'd be forced to provide that `defaultValue` to the hook explicitly and it becomes quite cumbersome. Also, at this point, why don't you just initialize that actor in some kind of a "lazy"/initializing/not ready state so its state could always be read reliably by `useActor`?

**Comment by Andarist** [comment]
Also, I've seen:
- some extra casting in your code. Do you have problems with typing this stuff correctly? I could help out with that
- some `any`s within the typegen metadata and even an empty string somewhere there. Were those generated by our tooling? Both look incorrect
- that you **mutate** `context` within `assign`. This is invalid and should be avoided - please use immutable updates. you should always return a new copy of the `context` from your `assign` calls.

**Comment by jsgiant** [comment]
@Andarist Thanks for your inputs, I just kept the sample code to let you know our use case. We are using the actor only after the actor ref is being added to the context and let me remind you that we're getting the issue only in specific devices with the below configuration.

### Device Config

- Model: Samsung Galaxy M31
- OS Type: Android
- Browsers: Chrome, Samsung Internet
- Android version: 11

Though it is constrained to a specific device we don't want to take it easy as it is very important for us to keep our application error-free for all sets of users.

Thanks.

**Comment by Andarist** [comment]
@jsgiant did you perhaps have a chance to look any further into this?

**Comment by jsgiant** [comment]
Hi @Andarist , Currently we're working on upgrading the XState in our code. If the issue still persists, I'll reopen the issue again then. Thanks for the quick response and awesome library :)
